spec/models/git_repository_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'spec_helper'
+
+describe GitRepository do
+  it "no url" do
+    expect { FactoryGirl.create(:git_repository, :dirname => "abc") }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
It's weird that a factory would raise an exception...that's not the intention of factories...

Instead, you are literally testing the validations on the model, so you should call that directly...

```ruby
describe ".create!" do
  it "with a missing url" do
    expect { described_class.create! }.to raise_error(ActiveRecord::RecordInvalid)
  end

  it "with an invalid url" do
    expect { described_class.create!(:url => "abc") }.to raise_error(ActiveRecord::RecordInvalid)
  end
end
```